From ebe7f074f6ad5fe5b76af6443fb056b1dbf9f106 Mon Sep 17 00:00:00 2001
From: Mrinmoy Ghosh <mrinmoy_g@hotmail.com>
Date: Tue, 19 Aug 2025 23:41:53 +0000
Subject: [PATCH] zebra : EVPN VXLAN MH extern mode: Update debug print for
 NDA_PROTOCOL

This is to update netlink debug print,
 to print 'protocol' i.e NDA_PROTOCOL field for netlink messages.

Signed-off-by: Mrinmoy Ghosh <mrinmoy_g@hotmail.com>
Signed-off-by: Mrinmoy Ghosh <mrghosh@cisco.com>
Signed-off-by: Tamer Ahmed <tamerahmed@microsoft.com>
(cherry picked from commit a4d7bb5623084d098e37f390fcad87f479683f09)
---
 zebra/debug_nl.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/zebra/debug_nl.c b/zebra/debug_nl.c
index e7297c762a..62f1c0b64c 100644
--- a/zebra/debug_nl.c
+++ b/zebra/debug_nl.c
@@ -572,6 +572,8 @@ const char *neigh_rta2str(int type)
 		return "MASTER";
 	case NDA_LINK_NETNSID:
 		return "LINK_NETNSID";
+	case NDA_PROTOCOL:
+		return "PROTOCOL";
 	default:
 		return "UNKNOWN";
 	}
@@ -1147,6 +1149,7 @@ static void nlneigh_dump(struct ndmsg *ndm, size_t msglen)
 	char bytestr[16];
 	char dbuf[128];
 	unsigned short rta_type;
+	uint8_t protocol = 0;
 
 #ifndef NDA_RTA
 #define NDA_RTA(ndm)                                                           \
@@ -1201,7 +1204,10 @@ next_rta:
 		vid = *(uint16_t *)RTA_DATA(rta);
 		zlog_debug("      %d", vid);
 		break;
-
+	case NDA_PROTOCOL:
+		protocol = *(uint8_t *)RTA_DATA(rta);
+		zlog_debug("      %d", protocol);
+		break;
 	default:
 		/* NOTHING: unhandled. */
 		break;
-- 
2.34.1

